dashboard/lib/contact_rollups_v2.rb

```diff
@@ -28,6 +31,19 @@ def self.execute_query_in_transaction(query)
     end
   end
 
+  # Execute a query in the dashboard database and returns query results.
+  # The query uses a Sequel or ActiveRecord connection depends on the current Rails environment.
+  def self.retrieve_query_results(query)
+    # @see comments in +execute_query_in_transaction+ method for explanation
+    # why we have to use ActiveRecord connection in a test environment.
+    if Rails.env.test?
+      ActiveRecord::Base.connection.exec_query(query)
+    else
+      # +Database#[]+ method actually returns a dataset and does not fetch records until needed.
```

Comment:
Maybe clarify this comment with more detail or a link about what a "dataset" is?

Reply:
I will clarify it as `Sequel::Dataset`